tools/nixpkgs-hammer

```diff
@@ -112,6 +116,65 @@ def run_external_checks(
     return dict(result)
 
 
+def run_fetchurl_checks(
+    fetchurl_calls: List[Dict[str, Any]], excluded_rules: List[str]
+) -> Dict[str, Report]:
+
+    if "out-of-repo-commit" in excluded_rules:
+        return {}
+
+    github_fetch = namedtuple("github_fetch", ["url", "owner", "repo", "sha"])
+
+    def github_commit_fetches() -> Iterable[github_fetch]:
+        for call in fetchurl_calls:
+            url = call.get("url", "")
+            if not isinstance(url, str):
+                continue
+            m1 = re.search(
+                r"github\.com/(.*)/(.*)/(?:archive|commit|pull/\d+/commits)/(.{40})\.",
+                url,
+            )
+            if m1 is not None:
+                yield (url,) + m1.groups()
+
+    def _fetch_data(url: str, owner: str, repo: str, sha: str) -> Tuple[str, str]:
+        branch_commit_url = f"https://github.com/{owner}/{repo}/branch_commits/{sha}"
```

Comment:
That is an undocumented path which returns html. It could change at any time and I think info should be added as a comment.

Reply:
Do you know if there's another endpoint that's documented? I suppose a graphql or something could work too.

Reply:
When I researched that a while ago I didn't find anything else.